rust/src/tests/mocks/aggregator.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{
+    aggregator::service::{Aggregator, ServiceHandle as InnerServiceHandle, ServiceRequests},
+    common::client::Credentials,
+};
+use bytes::Bytes;
+use std::{future::Future, pin::Pin};
+
+#[derive(Clone)]
+pub struct ServiceHandle(InnerServiceHandle);
+
+pub struct ByteAggregator {
+    weights: Vec<Bytes>,
+}
+
+impl ByteAggregator {
+    pub fn new() -> ByteAggregator {
+        ByteAggregator { weights: vec![] }
+    }
+}
+
+impl Aggregator for ByteAggregator {
+    type Error = ();
+
+    type AddWeightsFut = Pin<Box<dyn Future<Output = Result<(), ()>> + Send>>;
+    type AggregateFut = Pin<Box<dyn Future<Output = Result<Bytes, ()>> + Send>>;
+
+    fn add_weights(&mut self, weights: Bytes) -> Self::AddWeightsFut {
+        self.weights.push(weights);
+        Box::pin(async move { Ok(()) })
+    }
+
+    fn aggregate(&mut self) -> Self::AggregateFut {
```

Comment:
That may a bit too fancy. It would be nice to have an aggregator that returns something unambiguous, easily predictable and verifiable.

Perhaps `add_weights` could remain as it is and append the incoming bytes, and `aggregate` could sort the resulting vec. That way when we have tests where multiple participants upload weights at the same time, we can easily determine the expected value. If we have the participants upload different sequences of bytes with distinct values (like `[0, 0, 0, 0]`, `[1, 1, 1, 1]`, up to `[0xff, 0xff, 0xff, 0xff]`) the expectd result would be `[0, 0, 0, 0, 1, 1, 1, 1, ..., 16, 16, 16, 16]` for example.